stack_source_integration_test.go

```diff
@@ -66,3 +66,63 @@ func TestStackSourceCreateUploadAndRead(t *testing.T) {
 		require.Fail(t, "timed out waiting for stack source to be processed")
 	}
 }
+
+func TestStackSourceSpeculatives(t *testing.T) {
+	skipUnlessBeta(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	t.Cleanup(orgTestCleanup)
+
+	oauthClient, cleanup := createOAuthClient(t, client, orgTest, nil)
+	t.Cleanup(cleanup)
+
+	stack, err := client.Stacks.Create(ctx, StackCreateOptions{
+		Project: orgTest.DefaultProject,
+		Name:    "test-stack",
+		VCSRepo: &StackVCSRepoOptions{
+			Identifier:   "hashicorp-guides/pet-nulls-stack",
+			OAuthTokenID: oauthClient.OAuthTokens[0].ID,
+		},
+	})
+	require.NoError(t, err)
+
+	stackHCP, err := client.Stacks.Create(ctx, StackCreateOptions{
```

Comment:
Small nit, but I think `stackVCS` and `stack` might be more suitable variable names here.